iffy/iffy.go

```diff
@@ -23,68 +21,53 @@ type Tester struct {
 
 type Headers map[string]string
 
-type Call struct {
-	Name       string
-	Method     string
-	QueryStr   string
-	Body       string
-	headers    Headers
-	respObject interface{}
-	checkers   []Checker
-}
-
-func (c *Call) ResponseObject(respObject interface{}) *Call {
-	c.respObject = respObject
-	return c
-}
-
-func (c *Call) Headers(h Headers) *Call {
-	c.headers = h
-	return c
-}
-
-func (c *Call) Checkers(ch ...Checker) *Call {
-	c.checkers = ch
-	return c
-}
-
 type Checker func(r *http.Response, body string, respObject interface{}) error
 
+type TemplaterFunc func(s string) string
+
 // Tester
 
 func NewTester(t *testing.T, r http.Handler, calls ...*Call) *Tester {
 	return &Tester{
 		t:      t,
 		r:      r,
 		values: make(Values),
+		Calls:  calls,
 	}
 }
 
-func (t *Tester) AddCall(name, method, querystr, body string) *Call {
+func (t *Tester) AddCall(name, method, querystr string, body interface{}) *Call {
+	var bodyInterface Body
+	switch body.(type) {
+	case Body:
+		bodyInterface = body.(Body)
+	default:
```

Comment:
what about nil body? on a simple GET, for instance

Reply:
Added a `NoopBody` structure to handle this case. If content-type is empty, we are not inserting the content-type inside the request headers either (keeping the compat we had from before as well)